### PR TITLE
FeedbackSessionAttributes: isInGracePeriod: handle corner case #8569

### LINKED ISSUE #8569: FeedbackSessionAttributes: isInGracePeriod: handle corner case
This is the condition for `isOpened`:
https://github.com/TEAMMATES/teammates/blob/3e484a84ee695ce39e31f173ff7e674614fa3742/src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java#L287

This is the condition for `isInGracePeriod`:
https://github.com/TEAMMATES/teammates/blob/3e484a84ee695ce39e31f173ff7e674614fa3742/src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java#L301

There is an instant in time when `now` is exactly equal to `endTime` to the millisecond, causing both `isOpened` and `isInGracePeriod` to return `false` when the session is in fact transitioning between the two. I believe this to be the cause of random test failures involving testing the grace period, as the tests set the `endTime` to the current time then immediately test `isOpened` and `isInGracePeriod`:
https://github.com/TEAMMATES/teammates/blob/8555c6570219ae92e66a85f662abd73a0a726d4e/src/test/java/teammates/test/cases/action/InstructorFeedbackSubmissionEditSaveActionTest.java#L627-L631

Let's change the condition for `isInGracePeriod` to include the corner case when `now` is exactly equal to `endTime`.

### PR DESCRIPTION
**Pull request by whipermr5**
Fixes #8569

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java [hunk 1]
```diff
@@ -298,7 +298,7 @@ public boolean isInGracePeriod() {
         Date now = new Date();
         Date gracedEnd = new Date(endTime.getTime() + gracePeriod * 60000L);
 
-        return now.after(endTime) && now.before(gracedEnd);
+        return (now.after(endTime) || now.equals(endTime)) && now.before(gracedEnd);
```
Comment (whipermr5) [inline src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java]: I also considered `!now.before(endTime) && now.before(gracedEnd)`, but that might be slightly less readable.
